code/lib/preview-api/src/modules/preview-web/PreviewWithSelection.tsx

```diff
@@ -45,7 +45,7 @@ import type { StorySpecifier } from '../store/StoryIndexStore';
 const globalWindow = globalThis;
 
 function focusInInput(event: Event) {
-  const target = event.target as Element;
+  const target = event.composedPath()[0] as Element;
```

Comment:
I figured I'd leave this here for future me:
https://developer.mozilla.org/en-US/docs/Web/API/Event/composedPath

Reply:
FYI, I found this article to be really helpful on the topic of events and shadow dom
https://pm.dartus.fr/blog/a-complete-guide-on-shadow-dom-and-event-propagation/